tests: Add sysctl ipv6 and extra CI steps

lexnv opened the pull request:
Changes the binding in response to 
- https://github.com/docker/for-linux/issues/1374
This is achieved via `--sysctl net.ipv6.conf.all.disable_ipv6=0`


CI steps that are running now:
- cargo fmt
- cargo machete for unused dependencies (has detected one)
- cargo check
- cargo doc (with extra checks)
- cargo clippy
- cargo test

While at it, I've also fixed cargo doc tests.

I've added `reorder_imports` (similar to Substrate) to group imports from the same crate.

At the time being, the clippy step is allowed to fail to not block this PR further and a proper fix should land with: https://github.com/paritytech/litep2p/pull/57.

Builds on https://github.com/paritytech/litep2p/pull/56

alvicsam commented on .github/workflows/ci.yml:
```diff
@@ -40,6 +40,7 @@ jobs:
     container:
```
```suggestion
    timeout-minutes: 20
    container:
```
Can we also add time-out here? Because it seems that the job stuck and default timeout is 360 minutes

lexnv commented:
I think we could keep the `sysctl` fix, I'll revert my binding changes, good point thanks!

lexnv commented:
@dmitry-markin Let me know what you think of it, since I've change quite a few things.

CI steps that are running now:
- cargo fmt
- cargo machete for unused dependencies (has detected one)
- cargo check
- cargo doc (with extra checks)
- cargo clippy
- cargo test

While at it, I've also fixed cargo doc tests.

I've added `reorder_imports` (similar to Substrate) to group imports from the same crate.
Since these are quite opinionated, would love to get your input on them, we could otherwise revert the parts you think are unnecessary / too restrictive 🙏 

At the time being, the clippy step is allowed to fail to not block this PR further and a proper fix should land with: https://github.com/paritytech/litep2p/pull/57.

dmitry-markin commented on .github/workflows/ci.yml:
```diff
@@ -12,98 +12,144 @@ concurrency:
-      - name: doc
-        run: cargo doc
+      - name: Check documentation
+        run: RUSTDOCFLAGS="-D warnings -D rustdoc::broken_intra_doc_links" cargo doc --workspace --no-deps --document-private-items
```
Haven't checked,  `--document-private-items` does not generate an error if a private element is not documented, does it?

dmitry-markin commented on .github/workflows/ci.yml:
```diff
@@ -12,98 +12,144 @@ concurrency:
-  test:
-    name: "test"
+      - if: "failure()"
+        uses: "andymckay/cancel-action@271cfbfa11ca9222f7be99a47e8f929574549e0a" # v0.4
```
Does this cancel `Cargo fmt` check only, or all the CI checks? I would make checks to be independent.

lexnv replied:
This would cancel all running CI steps. We've been using it in subxt to reduce the CI minutes in case of failure.
However, I've removed this from litep2p since I believe we don't consume that many minutes 👍

dmitry-markin commented:
> I've added `reorder_imports` (similar to Substrate) to group imports from the same crate.
> Since these are quite opinionated, would love to get your input on them, we could otherwise revert the parts you think are unnecessary / too restrictive 🙏

Makes sense to me. Overall, you changes seem to be a big improvement, thank you!

One thing I don't understand / don't know how to check are all that github CI actions like:
```
uses: "andymckay/cancel-action@271cfbfa11ca9222f7be99a47e8f929574549e0a"
```
Not sure, how do we select the audited/trusted ones? (But may be if they work that's all we need to know.)

alvicsam commented on .github/workflows/ci.yml:
```diff
@@ -12,98 +12,144 @@ concurrency:
-      # - name: fmt
-      #   run: |
-      #     cargo fmt --all --check
+      - name: Install Rust nightly toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+          components: rustfmt
```
```suggestion
```
This part is not needed since we run the action in [the `ci-unified`docker image](https://github.com/paritytech/scripts/tree/master/dockerfiles/ci-unified). Also the `actions-rs` org is archived and the actions are not maintained anymore.
